src/test/java/codeu/model/store/basic/MessageStoreTest.java

```diff
@@ -14,84 +16,95 @@
+                    UUID.randomUUID(),
+                    "message two",
+                    Instant.ofEpochMilli(2000));
+    private final Message MESSAGE_THREE =
+            new Message(
+                    UUID.randomUUID(),
+                    UUID.randomUUID(),
+                    USER_ONE,
+                    "message three",
+                    Instant.ofEpochMilli(3000));
+
+    @Before
+    public void setup() {
+        mockPersistentStorageAgent = Mockito.mock(PersistentStorageAgent.class);
+        messageStore = MessageStore.getTestInstance(mockPersistentStorageAgent);
+
+        final List<Message> messageList = new ArrayList<>();
+        messageList.add(MESSAGE_ONE);
+        messageList.add(MESSAGE_TWO);
+        messageList.add(MESSAGE_THREE);
+        messageStore.setMessages(messageList);
+    }
+
+    @Test
+    public void testGetMessagesInConversation() {
+        List<Message> resultMessages = messageStore.getMessagesInConversation(CONVERSATION_ID_ONE);
+
+        Assert.assertEquals(2, resultMessages.size());
+        assertEquals(MESSAGE_ONE, resultMessages.get(0));
+        assertEquals(MESSAGE_TWO, resultMessages.get(1));
+    }
+
+    @Test
+    public void testAddMessage() {
+        UUID inputConversationId = UUID.randomUUID();
+        Message inputMessage =
+                new Message(
+                        UUID.randomUUID(),
+                        inputConversationId,
+                        UUID.randomUUID(),
+                        "test message",
+                        Instant.now());
+
+        messageStore.addMessage(inputMessage);
+        Message resultMessage = messageStore.getMessagesInConversation(inputConversationId).get(0);
+
+        assertEquals(inputMessage, resultMessage);
+        Mockito.verify(mockPersistentStorageAgent).writeThrough(inputMessage);
+    }
+
+    @Test
+    public void testGetTotalMessages() {
+        Integer resultSize = messageStore.countTotalMessages();
+        Integer expectedResult = 3;
+
+        Assert.assertEquals(expectedResult, resultSize);
+    }
+
+    @Test
+    public void testGetMessagesOfUser() {
+        List<Message> sentMessages = messageStore.getMessagesOfUser(USER_ONE);
+        int expectedSize = 2;
+
+        Assert.assertEquals(expectedSize, sentMessages.size());
+        Assert.assertEquals(MESSAGE_ONE, sentMessages.get(0));
+        Assert.assertEquals(MESSAGE_THREE, sentMessages.get(1));
+    }
+
+    private void assertEquals(Message expectedMessage, Message actualMessage) {
+        Assert.assertEquals(expectedMessage.getId(), actualMessage.getId());
+        Assert.assertEquals(expectedMessage.getConversationId(), actualMessage.getConversationId());
+        Assert.assertEquals(expectedMessage.getAuthorId(), actualMessage.getAuthorId());
+        Assert.assertEquals(expectedMessage.getContent(), actualMessage.getContent());
+        Assert.assertEquals(expectedMessage.getCreationTime(), actualMessage.getCreationTime());
+    }
```

Comment:
For MessageStoreTest.java, it looks like I modified the whole thing, but I just added USER_ONE in line 23 to have a distinct User who writes Message #1 and #3 (before all Messages were written by UUID.random() so by a random User) so that I could test getMessagesOfUser(USER_ONE).